src/contracts/AbstractARM.sol

```diff
@@ -0,0 +1,652 @@
+
+    /// @notice Calculates the amount of liquidity assets for a given amount of shares
+    function convertToAssets(uint256 shares) public view returns (uint256 assets) {
+        assets = (shares * totalAssets()) / totalSupply();
+    }
+
+    /// @notice Set the Liquidity Provider Controller contract address.
+    /// Set to a zero address to disable the controller.
+    function setLiquidityProviderController(address _liquidityProviderController) external onlyOwner {
+        liquidityProviderController = _liquidityProviderController;
+
+        emit LiquidityProviderControllerUpdated(_liquidityProviderController);
+    }
+
+    ////////////////////////////////////////////////////
+    ///         Performance Fee Functions
+    ////////////////////////////////////////////////////
+
+    /// @notice Owner sets the performance fee on increased assets
+    /// @param _fee The performance fee measured in basis points (1/100th of a percent)
+    /// 10,000 = 100% performance fee
+    /// 500 = 5% performance fee
+    function setFee(uint256 _fee) external onlyOwner {
+        _setFee(_fee);
+    }
+
+    /// @notice Owner sets the account/contract that receives the performance fee
+    function setFeeCollector(address _feeCollector) external onlyOwner {
+        _setFeeCollector(_feeCollector);
+    }
+
+    function _setFee(uint256 _fee) internal {
+        require(_fee <= FEE_SCALE, "ARM: fee too high");
+
+        // Collect any performance fees up to this point using the old fee
+        collectFees();
+
+        fee = SafeCast.toUint16(_fee);
+
+        emit FeeUpdated(_fee);
+    }
+
+    function _setFeeCollector(address _feeCollector) internal {
+        require(_feeCollector != address(0), "ARM: invalid fee collector");
+
+        feeCollector = _feeCollector;
+
+        emit FeeCollectorUpdated(_feeCollector);
+    }
+
+    /// @notice Transfer accrued performance fees to the fee collector
+    /// This requires enough liquidity assets in the ARM to cover the accrued fees.
+    function collectFees() public returns (uint256 fees) {
+        uint256 newAvailableAssets;
+        // Accrue any performance fees up to this point
+        (fees, newAvailableAssets) = _feesAccrued();
+
+        if (fee == 0) return fees;
+
+        require(fees <= IERC20(liquidityAsset).balanceOf(address(this)), "ARM: insufficient liquidity");
+
+        IERC20(liquidityAsset).transfer(feeCollector, fees);
+
+        // Save the new available assets back to storage less the collected fees.
+        lastAvailableAssets = SafeCast.toInt128(SafeCast.toInt256(newAvailableAssets) - SafeCast.toInt256(fees));
+
+        emit FeeCollected(feeCollector, fees);
+    }
+
+    /// @notice Calculates the performance fees accrued since the last time fees were collected
+    function feesAccrued() public view returns (uint256 fees) {
+        (fees,) = _feesAccrued();
+    }
+
+    function _feesAccrued() public view returns (uint256 fees, uint256 newAvailableAssets) {
```

Comment:
🟠 It's awkward to have this leading underbar method marked public.  We should do one of the following:

- Make this internal and make `feesAccrued()` external.
- Remove the underbar, keep it public, and remove the other feesAccrued() method.

Reply:
Option 1 was what I was intending to do. I've fixed. Thanks